### PR TITLE
refactor(Commit): refactor _prompt_commit_questions and fix some type…

### PR DESCRIPTION
**Pull request by bearomorphism**
… hint

## Checklist

- [ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly
- [ ] Check and fix any broken links (internal or external) in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: commitizen/commands/commit.py [hunk 1]
```diff
@@ -64,31 +64,35 @@ def _read_backup_message(self) -> str | None:
...
+    def _validate_subject_length(self, message: str) -> None:
+        # By the contract, message_length_limit is set to 0 for no limit
+        subject = message.partition("\n")[0].strip()
+        limit = self.arguments.get("message_length_limit", 0)
+        if limit == 0:
+            return
+        if len(subject) > limit:
             raise CommitMessageLengthExceededError(
-                f"Length of commit message exceeds limit ({message_len}/{message_length_limit})"
+                f"Length of commit message exceeds limit ({len(subject)}/{limit}), subject: '{subject}'"
```
Comment (bearomorphism) [inline commitizen/commands/commit.py]: I believe appending more details is not a breaking change?

File: commitizen/commands/commit.py [hunk 2]
```diff
@@ -64,31 +64,35 @@ def _read_backup_message(self) -> str | None:
...
-        message = cz.message(answers)
-        message_len = len(message.partition("\n")[0].strip())
-        message_length_limit = self.arguments.get("message_length_limit", 0)
-        if 0 < message_length_limit < message_len:
+        message = self.cz.message(answers)
+        self._validate_subject_length(message)
+        return message
+
+    def _validate_subject_length(self, message: str) -> None:
+        # By the contract, message_length_limit is set to 0 for no limit
```
Comment (bearomorphism) [inline commitizen/commands/commit.py:88]: Not sure if it still works if we use `None` for no limit

Comment (bearomorphism) [reply commitizen/commands/commit.py:88]: Related to a comment in #1574

File: commitizen/commands/commit.py (hunk 1 above)
Comment (Lee-W) [reply commitizen/commands/commit.py]: Yep, it should be fine

File: commitizen/commands/commit.py (hunk 2 above)
Comment (Lee-W) [reply commitizen/commands/commit.py:88]: If we use None, it's most likely to break, but should not be too hard to handle that

File: commitizen/commands/commit.py [hunk 3]
```diff
@@ -62,33 +62,37 @@ def _read_backup_message(self) -> str | None:
...
+        subject = message.partition("\n")[0].strip()
+        limit = self.arguments.get("message_length_limit", 0)
+        if limit == 0:
+            return
```
Comment (Lee-W) [inline commitizen/commands/commit.py]: This can be checked before running the `_validate_subject_length`. If there's nothing to validate, then we don't do validation

File: commitizen/commands/commit.py (hunk 2 above)
Comment (Lee-W) [reply commitizen/commands/commit.py:88]: will take a look at that PR later. But I'll get this merged

File: commitizen/commands/commit.py (hunk 3 above)
Comment (Lee-W) [reply commitizen/commands/commit.py]: Easy fix. I'll do it and push it. Thanks!

**Review by Lee-W — APPROVED** [review]
